src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1

```diff
@@ -446,6 +452,78 @@ class WinGetPackageManager
     }
 }
 
+[DSCResource()]
+class WinGetConfigRoot
+{
+    [DscProperty(Key)]
+    [string]$WinGetConfigRoot = "WinGetConfigRoot"
+
+    [DscProperty()]
+    [string]$Path = ""
+
+    [DscProperty()]
+    [bool]$Exist = $false
+
+    [DscProperty()]
+    [WinGetScope]$WinGetScope = [WinGetScope]::User
+
+    [WinGetConfigRoot] Get()
+    {
+        $currentState = [WinGetConfigRoot]::new()
+        $wingetConfigRootValue = [System.Environment]::GetEnvironmentVariable($this.WinGetConfigRoot, $this.WinGetScope)
+
+        if (-not [string]::IsNullOrEmpty($wingetConfigRootValue))
+        {
+            $currentState.Exist = $true
+            $currentState.Path = $wingetConfigRootValue
+        }
+
+        $currentState.WinGetScope = $this.WinGetScope
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        if ($this.Exist -ne $currentState.Exist)
+        {
+            return $false
+        }
+
+        if ($this.Path -ne $currentState.Path)
+        {
+            return $false
+        }
+
+        if ($this.WinGetScope -ne $currentState.WinGetScope)
+        {
+            return $false
+        }
+
+        if ($this.WinGetConfigRoot -ne $currentState.WinGetConfigRoot)
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+            if ($this.Exist)
+            {
+                [System.Environment]::SetEnvironmentVariable($this.WinGetConfigRoot, $this.Path, $this.WinGetScope)
+            }
+            else
+            {
+                [System.Environment]::SetEnvironmentVariable($this.WinGetConfigRoot, "", $this.WinGetScope)

```

Comment:
Nit: this could set the variable to `$null` to delete it instead of setting it to the empty string. According to the docs, `""` does not delete the variable since .NET 9 (no idea which one this uses...)